packages/server/src/routes/cron/notifyOnUnminted.ts

```diff
@@ -1,29 +1,134 @@
+  if (countryCode) {
+    const regionCode = phoneNumberUtil.getRegionCodeForCountryCode(countryCode);
+    const defaultTimezoneByCountryCode = require("default-timezone-for-country/data/default_timezone_by_country_code.json");
+    const timezone = defaultTimezoneByCountryCode[regionCode];
+    if (timezone) {
+      return timezone.utc_offset;
+    }
+  }
+  return undefined;
+}
+
+/**
+ * Return that we should notify the user if all the above is true:
+ * * their max-mintable is >= the mint camp.
+ * * they have an associated phone number (otherwise we can't notify them)
+ * * their inferred local time is within the noon hour
+ * * we haven't already notified them about their mintable amount
+ */
+async function shouldNotify(
+  auth: adminAuth.Auth,
+  member: DocumentSnapshot,
+  memberNotificationHistory: DocumentSnapshot
+) {
+  const maxMintable = calculateMaxMintableForMember(member);
+  if (maxMintable.gte(MINT_CAP)) {
+    const userRecord = await auth.getUser(member.id);
+    const phoneNumber = userRecord.phoneNumber;
+    if (phoneNumber) {
+      const offset = getDefaultOffsetForPhoneNumber(phoneNumber);
+      const localTime = moment().utcOffset(offset || 0);
+      if (localTime.hour() === HOUR_TO_NOTIFY) {
+        if (memberNotificationHistory.exists) {
+          return !memberNotificationHistory.get("notifiedOnUnminted");
+        }
+        return true;
+      }
+    }
+  }
+  return false;
+}
 
 export const notifyOnUnminted = (
+  auth: adminAuth.Auth,
   db: Firestore,
   messaging: adminMessaging.Messaging,
+  membersCollection: CollectionReference,
   fcmTokensCollection: CollectionReference,
-  membersCollection: CollectionReference
+  notificationHistoryCollection: CollectionReference
 ) =>
-  createApiRoute<CronNotifyOnUnmintedApiEndpoint>(
-    async (call, loggedInMemberToken) => {
-      sendPushNotification(
-        messaging,
-        fcmTokensCollection,
-        RAHUL_TEST_MEMBER_UID,
-        "This is a test.",
-        "Fear not."
-      );
-      return {
-        body: { message: "Test successful." },
-        status: 200
-      };
-    }
-  );
+  createApiRoute<CronNotifyOnUnmintedApiEndpoint>(async call => {
+    (await membersCollection.get()).forEach(async m => {
+      db.runTransaction(async transaction => {
+        try {
+          const member = await transaction.get(membersCollection.doc(m.id));
+          const memberNotificationHistory = await transaction.get(
+            notificationHistoryCollection.doc(member.id)
+          );
+          if (await shouldNotify(auth, member, memberNotificationHistory)) {
+            const title = `Mint ${calculateMaxMintableForMember(member)
+              .round(2, 0)
+              .toString()} Raha before it's too late!`;
+            const body =
+              "Your mintable amount will be capped at 40 Raha on November 15th. Tap this notification to mint now and avoid losing Raha!";
```

Comment:
11/15 or Nov. 15?

Reply:
hmm I guess 11/15 to be consistent

Reply:
Although, I think I prefer Nov 15. I find 11/15 hard to parse. I'll go with that.